lib/foreman_ansible_core/roles_reader.rb

```diff
@@ -40,6 +27,32 @@ def logger
           ::Proxy::LogBuffer::Decorator.instance
         end
       end
+
+      private
+
+      def read_roles(roles_path)
+        logger.info('[foreman_ansible] - Reading roles from '\
+                    "/etc/ansible/ansible.cfg #{roles_path}")
+        Dir.glob("#{roles_path}/*").map do |path|
+          path.split('/').last
+        end
+      rescue Errno::ENOENT, Errno::EACCES => e
+        logger.debug("[foreman_ansible] - #{e.backtrace}")
+        exception_message = '[foreman_ansible] - Could not read Ansible roles'\
+                            " from #{roles_path} - #{e.message}"
+        raise ReadConfigFileException.new(exception_message)
+      end
+
+      def roles_path_from_config
+        File.readlines(DEFAULT_CONFIG_FILE).select do |line|
+          line =~ /roles_path/
+        end
+      rescue Errno::ENOENT, Errno::EACCES => e
+        logger.debug("[foreman_ansible] - #{e.backtrace}")
```

Comment:
`logger.error(e)` would be better, both from backtrace formating, as well for error level

Reply:
oh, I see your passing different exception, is n this case, the original approach is probably ok